packages/components/src/popover/index.tsx

```diff
@@ -548,15 +563,26 @@ const UnforwardedPopover = (
 		</AnimatedWrapper>
 	);
 
-	if ( slot.ref ) {
+	const shouldRenderWithinSlot = slot.ref && ! inline;
+	const hasAnchor = anchorRef || anchorRect || anchor;
+
+	if ( shouldRenderWithinSlot ) {
 		content = <Fill name={ slotName }>{ content }</Fill>;
 	}
 
-	if ( anchorRef || anchorRect || anchor ) {
+	if ( ! hasAnchor ) {
+		content = <span ref={ anchorRefFallback }>{ content }</span>;
+	}
+
+	if ( shouldRenderWithinSlot ) {
+		return content;
+	}
+
+	if ( inline ) {
 		return content;
 	}
```

Comment:
Nit: could be a single condition:

```suggestion
	if ( shouldRenderWithinSlot || inline ) {
		return content;
	}
```

Reply:
Right. Sorry the PR was auto-merged. I'll open a follow up with some of these changes.